src/immer.js

```diff
@@ -131,6 +134,12 @@ export class Immer {
         if (!state.finalized) {
             state.finalized = true
             this.finalizeTree(state.draft, path, patches, inversePatches)
+            if (this.onDelete) {
+                const {assigned} = state
+                for (const prop in assigned)
+                    assigned[prop] || this.onDelete(state, prop)
```

Comment:
I think this should be `assigned[prop] === false && this.onDelete(state, prop)` (`undefined` means nothing happened with the prop from top of my head)

Reply:
The `in` operator protects against that. :)